pkg/controller/v1alpha1/localmodelnode/controller.go

```diff
@@ -67,12 +68,43 @@ var (
 	jobNamespace     string
 	nodeName         = os.Getenv("NODE_NAME") // Name of current node, passed as an env variable via downward API
 	modelsRootFolder = filepath.Join(MountPath, `models`)
-	removeAll        = os.RemoveAll // For patching os.RemoveAll in controller tests
-	readDir          = os.ReadDir   // For patching os.ReadDir in controller tests
+	fsHelper         fileSystemInterface
 )
 
-// Launch a new job or return an existing job
-func (c *LocalModelNodeReconciler) launchJob(ctx context.Context, jobName string, localModelNode *v1alpha1api.LocalModelNode, modelInfo v1alpha1api.LocalModelInfo, claimName string) (*batchv1.Job, error) {
+type fileSystemInterface interface {
+	removeAll(path string) error
+	readDir(name string) ([]os.DirEntry, error)
+	hasModelFolder(modelName string) (bool, error)
+}
+
+type fileSystemHelper struct {
+}
+
+func getModelFolder(modelName string) string {
+	return filepath.Join(modelsRootFolder, modelName)
+}
+
+func (f *fileSystemHelper) removeAll(path string) error {
+	return os.RemoveAll(path)
+}
+
+func (f *fileSystemHelper) readDir(path string) ([]os.DirEntry, error) {
+	return os.ReadDir(path)
+}
+
+func (f *fileSystemHelper) hasModelFolder(modelName string) (bool, error) {
+	folder := getModelFolder(modelName)
+	_, err := f.readDir(folder)
+	if err == nil {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+func (c *LocalModelNodeReconciler) getOrLaunchJob(ctx context.Context, jobName string, localModelNode *v1alpha1api.LocalModelNode, modelInfo v1alpha1api.LocalModelInfo, recreateJob bool) (*batchv1.Job, error) {
```

Comment:
I found it is  a bit odd to pass the `recreateJob` flag to a function and make the code harder to read.